README.md

```diff
@@ -1,32 +1,122 @@
+
+In order to achieve the above goal, a few sub-goals need to be met:
+* the controller needs to know about resource type events
+* the controller needs to be passed relevant details regarding the event to a handler
+* (optional) the controller __may__ want to interact with Kubernetes resource types
+
+There are many frameworks to help achieve the above goals. Lasso is one of them. What most of these frameworks have in common, including lasso, is that they use functionality from client-go to achieve these goals. You could achieve these goals by using just client-go. For the remainder of this section, we will fly through how a controller is put together just using client-go concepts.
+
+An informer is used to achieve the above goals needed for a Kubernetes Controller. An informer possesses an event handler and a set of indexers. The underlying structs are different but a pseudo struct is shown below to illustrate this:
+
+```
+Informer
+	EventHandler
+	Indexers
+```
+
+The EventHandler can have different logic for create, update, and delete events. The event handler is where the controller comes in. You use it to configure the create, update, and delete event handlers.
+
+The indexers are a type of computational cache. They are used to map a key to an object, given a function, then the result is saved. Here it is easiest to think of them simply- they receive some metadata such as the name and namespace of the an object and return that object.
+
+The informer first populates the indexer by listing it from a client. Then the informer watches the Kubernetes resource it is assigned to. When an event comes through, it uses that event to update its indexer (by remapping the object associated with the event) and to execute its event handlers.
+
+Usually an application would want multiple controllers. However, it may not be efficient for every controller to have its own informer and therefore its own indexer. The SharedIndexerInformer from client-go can be used to add multiple controllers that all use the same indexers.
+
+There is a great visual of how a Kubernetes controller might work found in the [kubernetes/sample-controller repo](https://github.com/kubernetes/sample-controller/blob/master/docs/images/client-go-controller-interaction.jpeg).
+
+__Note:__ Informers have been simplified for the sake of clarity. If you are interested in exactly how they work it is recommended you read through the [client-go code](https://github.com/kubernetes/client-go).
+
+# Lasso's Value Proposition
+### Lasso Controller
+Lasso standardizes what a controller is with its controller package. Lasso introduces factories for managing caches, controllers, and clients within a project. This is imperative for rancher which manages many instances of many types.
```

Comment:
I think it remains important to highlight that, in Kubernetes jargon, a controller is a pattern, while in Lasso it is an object.

The main pain point Lasso strives to solve is DX in writing controllers: make it as easy as implementing few interfaces, like normal practice in Golang.

Reply:
I went ahead and updated the opening definition of controller to clarify that it is lasso which uses it to refer to its controller structure(s).